src/Filters/atoms/DateTimeBlock.tsx

```diff
@@ -96,22 +105,32 @@ interface IProps {
   setDateCallback?: (date: Date) => void
   setTimeCallback?: (date: Date) => void
   allowAfterMidnight?: boolean,
+  checkTimeValidation?: boolean,
 }
 
 const DateTimeBlock : React.FC<IProps> = ({
   allowAfterMidnight = false,
   title,
   hasDate,
   hasTime,
+  checkTimeValidation,
   date = new Date(),
   setDateCallback = ()=>{},
 }) => {
 
-
-  const [displayHours, setDisplayHours] = useState<string>(format(date, "mm"));
-  const [displayMinutes, setDisplayMinutes] = useState<string>(format(date,'HH'));
+  const convertHours = (date.getHours()).toString();
+  const convertMinutes = date.getMinutes().toString();
+  const [displayHours, setDisplayHours] = useState<string>(convertHours);
+  const [displayMinutes, setDisplayMinutes] = useState<string>(convertMinutes);
 
   const setDateHours = useCallback(({target: {value}}: React.ChangeEvent<HTMLInputElement>) => {
+    const isContainsSymbolN = /^[0-1]{0,1}[0-9]$|(^2[0-4])$/;
+    if(Number(value)>24){
+      value = '24';
+    } else if (!isContainsSymbolN.test(value) && value !== '') {
+      return;
+    }
```

Comment:
Please reconsider this checks, I think you only need RegEx check.

Reply:
In regex we can't do this thing like..
if value is more that 24 like 55
it must add only 24 so I have added another `if`statement.
Please tell me your thoughts.